pysgrid/read_netcdf.py

```diff
@@ -124,7 +124,10 @@ def find_node_coordinates(self, node_dimensions):
             if x_node_coordinate is not None and y_node_coordinate is not None:
                 # exit the loop once both x and y coordinates are found
                 break
-        return x_node_coordinate, y_node_coordinate
+        if x_node_coordinate is not None and y_node_coordinate is not None:
```

Comment:
Question: Are there situations where one would have either `x_node_coordinate` or `y_node_coordinate` defined?  

If not, you might consider wrapping checkers, like the one above, into another function along these lines:

```python
def check_nodes(...):
    if isinstance(x_node_coordinate, None) and isinstance(y_node_coordinate, None):
        return False
    elif not isinstance(x_node_coordinate, None) and isinstance(y_node_coordinate, None):
        raise or warn for x_node_coordinate 
    elif isinstance(x_node_coordinate, None) and not isinstance(y_node_coordinate, None):
        raise or warn y_node_coordinate
    else:
        return True
```